src/components/ListLayout.jsx

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState, useRef } from 'react';
```

Comment:
A word of caution here. Obviously we're doing one PR per week so there isn't really a way around this, but ... in normal practice I would be careful about having a change like this happen amidst all the other changes. Here is why:

I'm not really sure how to read this diff. According to GitHub this is a brand new file, but what I (and every other reviewer can't really tell without _really_ getting into the weeds is: "is this file the exact same code that was in ListView.jsx OR did it also get some internal changes/optimizations? This is further made difficult because both teams are working in the same general files, so there will likely be some increased merge conflicts.

Again, there isn't really a way around this since we mostly do a PR per pair each week, but typically a refactor like this you would want to break out into it's own PR and have it change none of the logic, _just_ the code organization. That way the next PR could build off of the refactor one and you would be able to see all of the logic changes after the refactor.

Reply:
thanks for the review  @EdwardHinkle  , this makes a lot of sense, wasn't thinking that refactoring like this can cause such a problem, but now I can see this